### PR TITLE
traefik: support different repositories

### PR DESCRIPTION
**Pull request by macat**
Adding traefik repository variable so installations can overwrite it.

### CODE REVIEW

File: stable/traefik/values.yaml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Default values for Traefik
 imageTag: v1.1.1
```
Comment (prydonius) [inline stable/traefik/values.yaml]: can you remove the old one?

**Comment by prydonius** [comment]
Thanks a lot @macat! Can you also bump the chart version in Chart.yaml to `1.1.1-b`?

**Comment by prydonius** [comment]
@k8s-bot ok to test

File: stable/traefik/values.yaml (hunk 1 above)
Comment (macat) [reply stable/traefik/values.yaml]: Removed it. I wasn't sure how to handle values that are removed from a project.

**Comment by macat** [comment]
@prydonius Changed the chart version to `1.1.1-b`.

**Comment by viglesiasce** [comment]
cc @krancour to spot check this.

**Comment by krancour** [comment]
I would suggest two minor changes.

This PR changes `imageTag` to `image.tag`. That's a breaking change for anyone who might upgrade to this chart from the previous version whilst reusing their existing values.yaml (which they may keep in source control, for instance). I would advocate not changing that.

Keeping the above in mind, `image.repository` should probably be changed to `imageRepository`, but since that actually sounds redundant, I'd suggest further simplifying it to just `image`.

Sound ok?

**Comment by macat** [comment]
@krancour Would it make sense making the template smarter and checking if the `imageRepository` value is set? This way the chart could stay backwards compatible.

**Comment by krancour** [comment]
> Would it make sense making the template smarter and checking if the imageRepository value is set? This way the chart could stay backwards compatible.

I'm not sure what you're trying to achieve by doing that. Can you explain in more detail, please?

**Comment by krancour** [comment]
Bump. I think this is a valuable feature, but I still have my concerns over how it's implemented here. Can we either get this amended or otherwise close it?

**Comment by macat** [comment]
Yes, I agree. I started working on a possible solution but ran into a problem. I'll push my new implementation in a couple of days.

**Comment by viglesiasce** [comment]
Closing this out. @macat please submit a new PR when ready.

**Pull request closed without merging**